### DISCUSSION TITLE
Trying to do a POC using an iframe

### DISCUSSION THREAD

**Discussion by dubcekdao** [blockstack/stacks-wallet-web#2263] [Q&A]
I'm running into an issue doing a POC. I'm trying to run a nextjs app within an iframe. I'm able to fire the login popup, but the chrome extension isn't recognized as being installed.

First image is from web app running after clicking login.
<img width="1606" alt="Screen Shot 2022-02-23 at 10 50 22 PM" src="https://user-images.githubusercontent.com/100317441/155454608-b3896474-ea8d-4e3b-afd4-9ff97f7a9011.png">

Second image is same web app running within an iframe
<img width="1063" alt="Screen Shot 2022-02-23 at 10 51 39 PM" src="https://user-images.githubusercontent.com/100317441/155454733-4bf06a97-6abc-48ae-9719-db473133bf8f.png">

The goal is to be able to embed this app in an existing node code application to give users access to specific on chain functionality. Is this even possible? Are there security limitation?

**Comment by kyranjamie** [comment]
Trying to run the extension from an iframe isn't something I've considered. What is your objective, and why are you running a nextjs app in an iframe in the first place?

Web Extensions can specify an [`all_frames`](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#frames) option, so that this would work, though the Hiro Wallet doesn't currently support this.

I notice that Metamask _does_ use this setting.

**Reply by dubcekdao** [reply]
this was a way for me to reproduce the issue. I am trying to run a small app that interacts with a smart contract within a notion page. Notion allows you to embed an iframe.

**Reply by dubcekdao** [reply]
This may make it easier for developers to embed apps in existing web2 applications, unless there is another solution

**Reply by dubcekdao** [reply]
Is this something you're open to adding to the extension? I could maybe make a pr

**Reply by kyranjamie** [reply]
I've opened an issue to track this request.

**Reply by dubcekdao** [reply]
Thanks 👍
